### ISSUE TITLE
Format not working when lang is set to js in Vue files

### ISSUE THREAD

**Issue by kuoruan** [prettier/prettier#14910]
**Prettier 2.8.8**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmpaEAqc2+8AzmXgE5xkBmEbGB8ACasWAdzg4cAOhmkKFVAHpqpJZlwE4JKKqZg2ASywwy+KAHMAvMRAArJje044JgAocYMA3DZlL3Xn4TS0IyYHkyAE8IAFc2KXsACgByGAALWPM05IBKMgBfVUU9Q2NtXRhI520pPChhcLAmJiQyGAgIMgBCA2xeGDqTQqglJkrqkAAaEAhjA2gmZFB2NghRV3YERZQ8HFE8SMXpgCM2PDAAaxcAZTwMOAAZAyg4ZC5dpjgTs8ubrHPnuZkDA2DEviB6Mc4IJBNCHnVzDE8OY4AAxQICQHIEB4GLtKYgNIwDA4ADqaQMzH+YDg1y2lIMADdKZFsU0jiBnp82DB3Mj+G8PuD7AAPa6A5wARRiEHggpwn2m-zY3OxjLBBKwhlgpIMgnSyAAHAAGJWrT6ks5YbFauDcxmvaYARxl8Hcs22OKYAFoXtDoQSOC6DBw+eYBUh3grwZ8MAZgaCYxK4ABBTyGY54uCuHxPF7yxUgJjJ6Wy16RoXTAbHXX6tLIABMVbOBhwgIAwhAMBGIUwAKwEmKfGh4Y7bKOF9VwACS9QQMGu+iMMBT9Wu43LE7g+XyQA)

```sh
--parser vue
```

**Input:**

```vue
<template>
  <p>Templates are formatted as well...
    </p>
</template>

<script lang="js">
let Prettier = format => {    your.js('though') }
</script>

<style>
.and { css: too !important }
</style>
```

**Output:**

```vue
<template>
  <p>Templates are formatted as well...</p>
</template>

<script lang="js">
let Prettier = format => {    your.js('though') }
</script>

<style>
.and {
  css: too !important;
}
</style>

```

**Expected behavior:**
```vue
<template>
  <p>Templates are formatted as well...</p>
</template>

<script lang="js">
let Prettier = (format) => {
  your.js("though");
};
</script>

<style>
.and {
  css: too !important;
}
</style>
```

**Comment by auvred** [comment]
It seems that this issue has already been resolved in the `main` branch

[Playground link for the latest merged PR](https://deploy-preview-14900--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmpaEAqc2+8AzmXgE5xkBmEbGB8ACasWAdzg4cAOhmkKFVAHpqpJZlwE4JKKqZg2ASywwy+KAHMAvMRAArJje044JgAocYMA3DZlL3Xn4TS0IyYHkyAE8IAFc2KXsACgByGAALWPM05IBKMgBfVUU9Q2NtXRhI520pPChhcLAmJiQyGAgIMgBCA2xeGDqTQqglJkrqkAAaEAhjA2gmZFB2NghRV3YERZQ8HFE8SMXpgCM2PDAAaxcAZTwMOAAZAyg4ZC5dpjgTs8ubrHPnuZkDA2DEviB6Mc4IJBNCHnVzDE8OY4AAxQICQHIEB4GLtKYgNIwDA4ADqaQMzH+YDg1y2lIMADdKZFsU0jiBnp82DB3Mj+G8PuD7AAPa6A5wARRiEHggpwn2m-zY3OxjLBBKwhlgpIMgnSyAAHAAGJWrT6ks5YbFauDcxmvaYARxl8Hcs22OKYAFoXtDoQSOC6DBw+eYBUh3grwZ8MAZgaCYxK4ABBTyGY54uCuHxPF7yxUgJjJ6Wy16RoXTAbHXX6tLIABMVbOBhwgIAwhAMBGIUwAKwEmKfGh4Y7bKOF9VwACS9QQMGu+iMMBT9Wu43LE7g+XyQA)

**Comment by fisker** [comment]
Maybe fixed by #14439? Can add a test for it.

**Comment by kuoruan** [comment]
@fisker Done.
